### ISSUE TITLE
About train detection

### ISSUE THREAD

**Issue by IvenHsu** [gliese581gg/YOLO_tensorflow#1]
Hi gliese581gg, 

I follow this YOLO paper to write the detection layer,
but the output is not convergent.

I wonder if fc layer output values would influence the convergence status.

I tried to transform all the element in fc layer output to the range[0~1] with tensorflow API "tf.nn.sigmoid", 
but the detection layer out is still not convergent.

In your opinion, does the fc layer output (or other factors) matter to the convergence status?
Could you give some suggestions for me?

Thank you very much!

**Comment by gliese581gg** [comment]
Hi IvenHsu!
Since I'm not paper author, I can't tell you good answer.
but below is what i think.

1.
Are you trying to do training with my code?
I don't support training yet.

2.
I have not trained YOLO my self before. But in paper, they say they pretrained convolution layers with ImageNet for about a week. I think YOLO needs a lot of time for convergence like other deep convolutional neural nets.

3.
If you changed network's structure, the network will not converge if modified structure is not good enough or hyper parameters like learning rate is not properly set.

4.
Training YOLO needs complex dataset augmentation since YOLO uses unique label vectors.
You need to do that to your dataset before training

5.
applying sigmoid to final output will ruin everything.
YOLO's output layer is designed to predict exact values of boundingbox's location and class probability.

6.
How about visit (https://groups.google.com/forum/#!forum/darknet)
It is google forum for YOLO run by paper author.
YOLO's original authors may have answer to  your question.

**Comment by IvenHsu** [comment]
Appreciated your time and effort.
Your suggest is very helpful to me.
I implement the new Op of YOLO detection and train it. 
I will follow those suggest to find out the problem
Thanks very much.

**Comment by kcalcote** [comment]
IvenHsu, do you have and are you willing to share your training code?

**Comment by wangxm345566462** [comment]
IvenHsu, do you have and are you willing to share your training code?
